sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -240,6 +240,15 @@ object SQLConf {
     .intConf
     .createWithDefault(100)
 
+  val MULTI_ADD_OPT_THRESHOLD =
+    buildConf("spark.sql.analyzer.canonicalization.multiAddMemoryOptThreshold")
```

Comment:
shall we apply it to similar operations like `Multiply`? seems we can make changes in `trait CommutativeExpression`